packages/react/src/components/DesignLibrary/DesignLibraryClientEvents.tsx

```diff
@@ -70,31 +73,33 @@ export const DesignLibraryVariantGenerationEvents = ({
   previewComponentData,
 }: DesignLibraryVariantGenerationEventsProps) => {
   useEffect(() => {
+    if (!component?.uid) return;
+
     postToDesignLibrary(getDesignLibraryStatusEvent(designLibraryStatus, component.uid));
 
     const unsubUpdate = addComponentUpdateHandler(component, (updated) => {
       _updateServerComponentAction({
-        uid: updated.uid,
+        uid: updated.uid!,
         updatedComponent: updated,
         previewComponent: previewComponentData,
       });
     });
 
     const unsubPreview = addServerComponentPreviewHandler((eventArgs) => {
-      _updateServerComponentAction({ uid: component.uid, previewComponent: eventArgs });
+      _updateServerComponentAction({ uid: component.uid!, previewComponent: eventArgs });
     });
 
     if (importMapError) {
       // an import map error occurred on the server side in DesignLibraryServer, post error event to Design Studio
       sendErrorEvent(component.uid, importMapError, codegen.DesignLibraryPreviewError.RenderInit);
     } else {
-      const importMapEvent = getDesignLibraryImportMapEvent(component.uid, importMap);
+      const importMapEvent = getDesignLibraryImportMapEvent(component.uid, importMap!);
       postToDesignLibrary(importMapEvent);
 
       const propsEvent = getDesignLibraryComponentPropsEvent(
         component.uid,
-        component.fields,
-        component.params
+        component.fields || {},
+        component.params || {}
```

Comment:
See my comment regarding `getDesignLibraryComponentPropsEvent` function, we can make these parameters optional there and apply default value